tests/metrics/test_stratification.py

```diff
@@ -1,371 +1,252 @@
-        categories={str(i + 1) for i in range(12)},
-        mapper=lambda row: str(row[0].month),
-        current_categories_getter=lambda: {str(mock_clock().month)},
-    ),
-}
 
+class FavoriteColor:
 
-def unmock_instance_method(instance, method: Callable) -> Callable:
-    return lambda *args: method(instance, *args)
+    OPTIONS = ["red", "green", "orange"]
 
+    @property
+    def name(self) -> str:
+        return "favorite_color"
 
-@pytest.fixture
-def mock_stratification_level_init(mocker):
-    return mocker.patch(
-        "vivarium_public_health.metrics.stratification.StratificationLevel.__init__",
-        return_value=None,
-    )
+    def setup(self, builder: Builder):
+        self.randomness = builder.randomness.get_stream(self.name)
+        # Our simulants are fickle and change their
+        # favorite colors every time step.
+        builder.value.register_value_producer(
+            self.name,
+            requires_streams=[self.name],
+            source=lambda index: self.randomness.choice(index, choices=self.OPTIONS),
+        )
 
 
-@pytest.fixture()
-def mock_stratifier(mocker) -> MagicMock:
-    stratifier = mocker.MagicMock()
-    stratifier.setup = mocker.MagicMock(
-        side_effect=unmock_instance_method(stratifier, ResultsStratifier.setup)
-    )
-    return stratifier
+class FavoriteNumber:
 
+    OPTIONS = [7, 42, 14312]
 
-@pytest.fixture()
-def mock_builder(mocker) -> MagicMock:
-    builder = mocker.MagicMock()
+    @property
+    def name(self) -> str:
+        return "favorite_number"
 
-    builder.configuration.observers.default = DEFAULT_CONFIGS
-    builder.configuration.time.start.year = START_YEAR
-    builder.configuration.time.end.year = END_YEAR
+    def setup(self, builder: Builder) -> None:
+        self.randomness = builder.randomness.get_stream(self.name)
+        self.population_view = builder.population.get_view([self.name])
+        builder.population.initializes_simulants(
+            self.on_initialize_simulants,
+            creates_columns=[self.name],
+            requires_streams=[self.name],
+        )
 
-    builder.data.load = mocker.MagicMock(side_effect=load)
-    builder.value.get_value = mocker.MagicMock(side_effect=get_value)
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        # Favorite numbers are forever though.
+        self.population_view.update(
+            self.randomness.choice(
+                pop_data.index,
+                choices=self.OPTIONS,
+            ).rename(self.name)
+        )
 
-    builder.time.clock.return_value = mock_clock
-    return builder
 
+FAVORITE_THINGS = {
+    f"{color}_{number}"
+    for color, number in itertools.product(FavoriteColor.OPTIONS, FavoriteNumber.OPTIONS)
+}
 
-def mock_clock() -> datetime:
-    return datetime(2022, 5, 3, 12)
 
+class ResultsStratifier(ResultsStratifier_):
+    def register_stratifications(self, builder: Builder) -> None:
```

Comment:
Here is another example. We are doing what we'd do in actual practice -- extending the results stratifier to add new startifications.  Test suites can often serve as example use cases when written well.